### PR TITLE
Migrate loadChannelsForCurrentUser action to redux

### PR DESCRIPTION
**Pull request by enahum**
@saturninoabril let me know if there are more tests that I can add to this, I did try to do a bunch of stuff to test the inner functions and inner dispatches but I just wasn't able to

Affected areas: Everywhere that shows the channels of the user (eg. LHS) Ctrl+K, etc.. also when loading a permalink in a different window

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12702

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by enahum** [comment]
@DHaussermann does seem to be valid bugs but not related to this PR, can you make sure they also happen in master? cause if they do definitely not related to this PR

**Comment by DHaussermann** [comment]
@enahum I have now reported both bugs. Issue 1 was old. For issue 2 with leaving channels I reported https://mattermost.atlassian.net/browse/MM-12883 but the behavior is different from this SpinMint.

**Comment by DHaussermann** [comment]
Hi @enahum, The different build of Master would explain it. Is it time consuming to re-base it? I would appreciate this since this way, I could be sure that there are no new issues that I have not already reported against master.

**Comment by DHaussermann** [comment]
@enahum this now looks much more like the bug reported on master with the "Loading..." 

With alot of patience I was also able to repo the.  "TypeError: Cannot read property 'delete_at' of undefined" on pre-release.

Neither of these 2 issues was new in this SpinMint. Based on the testing done (outlined in MM-12702), this looks good to me.

File: tests/redux/actions/channel_actions.test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    test('loadChannelsForCurrentUser', async () => {
+        const testStore = await mockStore(initialState);
+
+        const expectedActions = [{
+            type: 'MOCK_FETCH_CHANNELS_AND_MEMBERS',
+            args: ['team-id'],
+        }];
+
+        await testStore.dispatch(Actions.loadChannelsForCurrentUser());
+        expect(testStore.getActions()).toEqual(expectedActions);
```
Comment (saturninoabril) [inline tests/redux/actions/channel_actions.test.js]: I also don't see much to test, maybe add whether the `loadProfilesForSidebar` has been called.
```javascript
import {loadProfilesForSidebar} from 'actions/user_actions.jsx';
...
expect(loadProfilesForSidebar).toHaveBeenCalledTimes(1);
```

File: actions/channel_actions.jsx [hunk 2]
```diff
@@ -64,36 +64,43 @@ export async function openGroupChannelToUsers(userIds, success, error) {
...
+        const state = getState();
+
+        await dispatch(ChannelActions.fetchMyChannelsAndMembers(getCurrentTeamId(state)));
+        dispatch(loadDMsAndGMsForUnreads());
+
+        loadProfilesForSidebar();
+    };
 }
 
 export function loadDMsAndGMsForUnreads() {
```
Comment (saturninoabril) [inline actions/channel_actions.jsx]: I'm not sure if this should be an action as we're not using `dispatch` internally and we're not calling `loadDMsAndGMsForUnreads` anywhere else.  Maybe just combine to `loadChannelsForCurrentUser`?

**Comment by enahum** [comment]
@saturninoabril changes made

**Pull request merged**